R/utils.R

```diff
@@ -1,21 +1,32 @@
-library(ggplot2)
-
 #' This is a title for a function
 #' @description This is a function here's where we describe what it does
 #' @param parameter Here's a parameter let's describe it here
+#' @import ggplot2
+#' @import kableExtra
 #' @export
 #' @examples \dontrun{
 #'
 #' pg_data <- example_data()
 #' }
 example_data <- function() {
+  ex_data <- list()
   file <- list.files(
     pattern = "PP_pgPEN_HeLa_counts.txt",
     recursive = TRUE,
     system.file("extdata", package = "gimap"),
     full.names = TRUE
   )
-  readr::read_tsv(file)
+  ex_data$countdata <- readr::read_tsv(file)
+  
+  file <- list.files(
```

Comment:
Can you explain this part?

Reply:
How do you envision this being used? Is this returning one big list of all the example data? We might instead want to allow the user to choose which example data they want. I have a set up like this on metricminer we could use https://github.com/fhdsl/metricminer/blob/4df22f1b1ec641d33ebb666811afee410a7cb744/R/utils.R#L45

Reply:
Because the input data is going to require counts and metadata, yes I'm envisioning a named list of all the example data where the documentation will describe what each part is. I'd started to fill out the vignette with this. They can't really choose which example data they want since it's all required. So if you want it to be an argument where they chose/specify which of the required examples to return and the function returns just that data, and not a named list, and then they rerun it, yes I can switch it to that.

Reply:
Yeah, makes sense. I'll make the change